bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkImpl.java

```diff
@@ -0,0 +1,100 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2021 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.link;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import com.adobe.cq.wcm.core.components.commons.link.Link;
+import com.day.cq.wcm.api.Page;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * Wraps link information to be used in models.
+ */
+@JsonInclude(Include.NON_NULL)
+public final class LinkImpl implements Link {
+
+    private final String url;
+    private final Map<String, String> htmlAttributes;
+    private final Page targetPage;
+
+    /**
+     * @param url Link URL
+     */
+    public LinkImpl(String url) {
+        this(url, null, null);
+    }
+
+    /**
+     * @param url Link URL
+     * @param target Target
+     */
+    public LinkImpl(String url, String target) {
+        this(url, target, null);
+    }
+
+    /**
+     * @param url Link URL
+     * @param target Target
+     * @param targetPage Target page
+     */
+    LinkImpl(String url, String target, Page targetPage) {
```

Comment:
I find this constructor confusing. We could have a `url` pointing to `page1` and `targetPage` pointing to `page2`. Where would the link point to in this case?
I'd rather introduce following constructors:
```
LinkImpl(Page targetPage)
LinkImpl(Page targetPage, String target)
```
which would mirror the other 2 constructors.

Reply:
the point is: link objects can represent any type of links - link to target pages, link to assets, link to external URLs.
the target page is just an optional parameter given as extra metadata together with an internal link.

Reply:
Since `LinkImpl` is only a POJO and the logic is in `LinkHandler` it makes sense to have this.